supervisor/bootstrap.py

```diff
@@ -256,8 +256,8 @@ def migrate_system_env(coresys: CoreSys) -> None:
             coresys.config.addons_repositories
         )
         coresys.config._data[ATTR_ADDONS_CUSTOM_LIST] = []
-        coresys.store.save_data()
-        coresys.config.save_data()
+        await coresys.store.save_data()
+        await coresys.config.save_data()
```

Comment:
Have you considered leaving `read_data`/`save_data` a regular function with blocking I/O, and use additional helper functions (like `load_config` you've already introduced) instead? I think this would allow to group I/O into a single executor call maybe. :thinking:

Reply:
So the tricky part here is even `save_data` can actually modify `self._data`. That's effectively a global in all these classes so modifying it doesn't seem like something we should do in the executor.

I think bootstrap is the main place we would want to leverage this anyway, basically anywhere else is just reading or writing data from one object. I had actually planned to come back to bootstrap at some point already. I'd like to do some of these in parallel with `asyncio.gather` if possible but didn't want to do that in this PR since there could be dependencies between them. I can take a look at how to group some of these then by refactoring into helpers.